assets/js/modules/analytics/components/common/UseUAandGA4SnippetSwitches.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Analytics Use UA and GA4 Snippet Switch component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { MODULES_ANALYTICS_4 } from '../../../analytics-4/datastore/constants';
+import Switch from '../../../../components/Switch';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function UseUAandGA4SnippetSwitches() {
+	const useUASnippet = useSelect( ( select ) => select( STORE_NAME ).getUseSnippet() );
+	const canUseUASnippet = useSelect( ( select ) => select( STORE_NAME ).getCanUseSnippet() );
+	const useGA4Snippet = useSelect( ( select ) => select( MODULES_ANALYTICS_4 ).getUseSnippet() );
+
+	const { setUseSnippet: setUseUASnippet } = useDispatch( STORE_NAME );
+	const { setUseSnippet: setUseGA4Snippet } = useDispatch( MODULES_ANALYTICS_4 );
+
+	const onUAChange = useCallback( () => {
+		setUseUASnippet( ! useUASnippet );
+		trackEvent( 'analytics_setup', useUASnippet ? 'analytics_tag_enabled' : 'analytics_tag_disabled' );
+	}, [ useUASnippet, setUseUASnippet ] );
+
+	const onGA4Change = useCallback( () => {
+		setUseGA4Snippet( ! useGA4Snippet );
```

Comment:
Looks like this is missing a `trackEvent` call, though I'll admit I'm not sure what it should be.

@felixarntz I assume we should add an event here, would:

```
trackEvent( 'analytics_setup', useUASnippet ? 'analytics4_tag_enabled' : 'analytics4_tag_disabled' );
```

make sense?

Reply:
@tofumatt Good catch! Yes, let's use these event names you're suggesting. cc @johnPhillips

Reply:
Thanks, I meant to ask about this. 👍